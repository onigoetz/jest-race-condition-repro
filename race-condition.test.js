/**
 * The Dummy code to test
 * Just a function that returns the first argument after 600ms
 * @param {*} content
 */
function longRunningFunction(content) {
  return new Promise((resolve, reject) => {
    setTimeout(() => {
      resolve(content);
    }, 600);
  });
}

// This is only needed in the test repro
// If the snapshots don't exist, leave the time for them to be created
// If they exist, fail the test on purpose
const fs = require("fs");
const path = require("path");
const snapshotFile = `${path.basename(__filename)}.snap`;
const snapshotPath = path.dirname(process.cwd(), "__snapshots__", snapshotFile);
if (fs.existsSync(snapshotPath)) {
  jest.setTimeout(400);
}

// The tests
it("Runs a first test", async () => {
  expect(
    await longRunningFunction("Content of the first test")
  ).toMatchSnapshot();
});

it("Runs a second test", async () => {
  expect(
    await longRunningFunction("Content of the second test")
  ).toMatchSnapshot();
});
