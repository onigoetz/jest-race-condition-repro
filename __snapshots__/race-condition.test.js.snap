// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Runs a first test 1`] = `"Content of the first test"`;

exports[`Runs a second test 1`] = `"Content of the second test"`;
